cryptoki/tests/basic.rs

```diff
@@ -1028,6 +1029,57 @@ fn aes_gcm_with_aad() -> TestResult {
     Ok(())
 }
 
+#[test]
+#[serial]
+fn rsa_pkcs_oaep_empty() -> TestResult {
+    let (pkcs11, slot) = init_pins();
+    let session = pkcs11.open_rw_session(slot)?;
+    session.login(UserType::User, Some(&AuthPin::new(USER_PIN.into())))?;
+
+    let pub_key_template = [Attribute::ModulusBits(2048.into())];
+    let (pubkey, privkey) =
+        session.generate_key_pair(&Mechanism::RsaPkcsKeyPairGen, &pub_key_template, &[])?;
+    let oaep = PkcsOaepParams::new(
+        MechanismType::SHA1,
+        PkcsMgfType::MGF1_SHA1,
+        PkcsOaepSource::empty(),
+    );
+    let encrypt_mechanism: Mechanism = Mechanism::RsaPkcsOaep(oaep);
+    let encrypted_data = session.encrypt(&encrypt_mechanism, pubkey, b"Hello")?;
+
+    let decrypted_data = session.decrypt(&encrypt_mechanism, privkey, &encrypted_data)?;
+    let decrypted = String::from_utf8(decrypted_data)?;
+    assert_eq!("Hello", decrypted);
+
+    Ok(())
+}
+
+#[test]
+#[serial]
+#[ignore] // it's not clear why the test with data specified fails
+fn rsa_pkcs_oaep_with_data() -> TestResult {
+    let (pkcs11, slot) = init_pins();
+    let session = pkcs11.open_rw_session(slot)?;
+    session.login(UserType::User, Some(&AuthPin::new(USER_PIN.into())))?;
+
+    let pub_key_template = [Attribute::ModulusBits(2048.into())];
+    let (pubkey, privkey) =
+        session.generate_key_pair(&Mechanism::RsaPkcsKeyPairGen, &pub_key_template, &[])?;
+    let oaep = PkcsOaepParams::new(
+        MechanismType::SHA1,
+        PkcsMgfType::MGF1_SHA1,
+        PkcsOaepSource::data_specified(&[1, 2, 3, 4, 5, 6, 7, 8]),
```

Comment:
We've had a similar problem in Parsec with the software TPM, where the salt/label would not be accepted unless it ends with a `0` byte. Maybe you can try and see if that helps?

Reply:
Hmm... I did some testing and browsed softhsm's source code and apparently they don't support any source data that is not null :/ (I'm on mobile now and will send links later).

Not sure if in that case it's better to remove the test case or leave it so that at least it's compile checked...?

Reply:
I think we can leave it for now! Good to know that we at least have it, and why it doesn't work on CI.

Reply:
Oh, it seems I forgot to update this PR with my findings. :sweat_smile: 

Here's a line of code in softhsm that checks `pSourceData` and returns `CKR_ARGUMENTS_BAD` if it is set (!= `NULL`):

https://github.com/opendnssec/SoftHSMv2/blob/f4661af6680e187f220921c0a91d17d71a680345/src/lib/SoftHSM.cpp#L12850

So, currently this is untestable with softhsm. Apparently it *is* supported by yubico library (ykcs11): https://github.com/Yubico/yubico-piv-tool/blob/a72b2054de4b8ee0bf534edadd4de64d0ded92a9/ykcs11/mechanisms.c#L983 I'll add testing that on a yubico to my ever-growing list of tasks to do :)

Thanks for the review and merging :wave: